admin-dev/themes/new-theme/js/pages/product/components/categories/index.js

```diff
@@ -0,0 +1,234 @@
+
+    if (hasChilds) {
+      listItem.classList.add('more');
+    }
+
+    const labelElement = document.createElement('label');
+
+    const inputElement = document.createElement('input');
+    inputElement.setAttribute('type', 'checkbox');
+    inputElement.setAttribute('name', 'form[step1][categories][tree][]');
+    inputElement.setAttribute('value', category.id);
+    inputElement.classList.add('category');
+
+    const radioElement = document.createElement('input');
+    radioElement.setAttribute('type', 'radio');
+    radioElement.setAttribute('name', 'ignore');
+    radioElement.setAttribute('value', category.id);
+    radioElement.classList.add('default-category');
+
+    labelElement.append(inputElement, ` ${category.name}`, radioElement);
+    checkboxContainer.append(labelElement);
+    listItem.append(checkboxContainer);
+
+    if (hasChilds) {
+      const childList = document.createElement('ul');
+      childList.classList.add('hide', 'child-list');
+
+      checkboxContainer.addEventListener('click', event => {
+        if (!event.target.classList.contains('default-category') && !event.target.classList.contains('category')) {
+          childList.classList.toggle('hide');
+
+          if (childList.classList.contains('hide')) {
+            checkboxContainer.parentElement.classList.remove('less');
+            checkboxContainer.parentElement.classList.add('more');
+          } else {
+            checkboxContainer.parentElement.classList.remove('more');
+            checkboxContainer.parentElement.classList.add('less');
+          }
+        }
+      });
+
+      category.childs.forEach(element => {
+        const child = this.createItem(element);
+
+        childList.append(child);
+      });
+
+      listItem.append(childList);
+    }
+
+    return listItem;
+  }
+
+  /**
+   * @param force {boolean} Force expanding instead of toggle
+   *
+   * Expand the category tree
+   */
+  toggleExpand(force) {
+    const currentAction = this.actionButton.querySelector(ProductCategoryMap.currentAction);
+    const nextAction = this.actionButton.querySelector(ProductCategoryMap.nextAction);
+    nextAction.classList.remove('hide');
+    nextAction.style.display = 'block';
+    currentAction.classList.add('hide');
+    currentAction.style.display = 'none';
+
+    this.categoriesElement.querySelectorAll(ProductCategoryMap.childList).forEach(e => {
+      if (this.expanded && !force) {
+        e.classList.add('hide');
+      } else {
+        e.classList.remove('hide');
+      }
+    });
+
+    this.categoriesElement.querySelectorAll(ProductCategoryMap.everyItems).forEach(e => {
+      if (this.expanded && !force) {
+        e.classList.add('more');
+        e.classList.remove('less');
+      } else {
+        e.classList.add('less');
+        e.classList.remove('more');
+      }
+    });
+
+    this.expanded = !this.expanded;
+  }
+
+  /**
+   * Typehead datas require to have only one array level
+   */
+  initTypeheadDatas(datas) {
```

Comment:
If Im not mistaken and this "typehead" is related to a search, then should be "Type**a**head" everywhere, shoudn't it?

Reply:
Oh damn, my bad

Reply:
Fixed it